src/shims/foreign_items.rs

```diff
@@ -138,7 +139,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             Entry::Occupied(e) => e.into_mut(),
             Entry::Vacant(e) => {
                 // Find it if it was not cached.
-                let mut instance_and_crate: Option<(ty::Instance<'_>, CrateNum)> = None;
+                let mut instance_and_crate: Option<(ty::Instance<'_>, CrateNum, bool)> = None;
```

Comment:
Please explain what the `bool` means, or use a self-explaining type. It might be worth using a struct instead of a tuple.